sermant-plugins/sermant-springboot-registry/springboot-registry-service/src/main/java/io/sermant/discovery/service/lb/rule/NacosWeightRandomRule.java

```
@@ -0,0 +1,95 @@
+public class NacosWeightRandomRule extends AbstractLoadbalancer {
+    @Override
+    protected ServiceInstance doChoose(String serviceName, List<ServiceInstance> instances) {
+        List<InstanceWithWeight> withWeights = instances.stream()
+                .map(instance -> {
+                    int weight = Integer.parseInt(instance.getMetadata().get("nacos.weight"));
+                    return new InstanceWithWeight(instance, weight);
+                }).collect(Collectors.toList());
+        return this.weightRandom(withWeights);
+    }
+
+    @Override
+    public String lbType() {
+        return "NacosWeight";
+    }
+
+    /**
+     * nacos Weight
+     *
+     * @author xiaozhao
+     * @since 2024-11-16
+     */
+    public static class InstanceWithWeight {
+        private ServiceInstance server;
+        private Integer weight;
+
+        /**
+         * constructor
+         *
+         * @param instance instance
+         * @param weight weight
+         */
+        public InstanceWithWeight(ServiceInstance instance, int weight) {
+            this.server = instance;
+            this.weight = weight;
+        }
+
+        public ServiceInstance getServer() {
+            return server;
+        }
+
+        public Integer getWeight() {
+            return weight;
+        }
+    }
+
+    /**
+     * Random by weight
+     *
+     * @param list instance list
+     * @return just a random result
+     */
+    public ServiceInstance weightRandom(List<InstanceWithWeight> list) {
+        int totalWeight = list.stream().mapToInt(InstanceWithWeight::getWeight).sum();
+        int randomWeight = new Random().nextInt(totalWeight);
+        int currentWeight = 0;
+        for (InstanceWithWeight instanceWithWeight : list) {
+            currentWeight += instanceWithWeight.getWeight();
+            if (randomWeight < currentWeight) {
+                return instanceWithWeight.getServer();
+            }
+        }
+        return list.get(0).getServer();
```

This is actually unreachable. Should we throw an exception or print a log?